plugins/inputs/apcupsd/apcupsd.go

```diff
@@ -0,0 +1,94 @@
+package apcupsd
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/mdlayher/apcupsd"
+)
+
+const defaultAddress = "127.0.0.1:3551"
+
+var defaultTimeout = internal.Duration{Duration: time.Duration(time.Second * 5)}
+
+type ApcUpsd struct {
+	Servers []string
+	Timeout internal.Duration
+}
+
+func (_ *ApcUpsd) Description() string {
+	return "Monitor APC UPSes connected to apcupsd"
+}
+
+var sampleConfig = `
+  # a list of running apcupsd server to connect to. 
+  # If not provided will default to 127.0.0.1:3551
+  servers = ["127.0.0.1:3551"]
```

Comment:
Sorry I didn't mention this sooner, but let's make this a URL: `"tcp://127.0.0.1:3551"` and use `url.Parse` to split it up.  This will allow ipv6 support and we won't have to adjust it later (something I have been doing a lot of recently).